src/Utils.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import Constants from "./Constants";
+import { PathUtils } from "./PathUtils";
+import MetaProperty from "./MetaProperty";
+import { ModelManager } from "./ModelManager";
+
+export class Utils {
+    /**
+     * Returns all the tags for requested state
+     *
+     * @returns {String} all the tags for requested state
+     */
+    public static getTagsForState(state : string) : string {
+        let tags = '';
+        if (state === Constants.AUTHORING) {
+            const clientLibs = this.generateClientLibsUrl();
+            clientLibs.forEach(clientlib => {
+                if (clientlib.endsWith('.js')) {
+                    tags = tags.concat(this.generateElementString('script', 'src', clientlib));
+                }
+                else if (clientlib.endsWith('.css')) {
+                    tags = tags.concat(this.generateElementString('stylesheet', 'href', clientlib));
+                }
+            });
+        }
+        return tags;
+    }
+
+    /**
+     * Returns string value of all the concatenated tags
+     *
+     * @returns {String} concatenated tags
+     */
+    private static generateElementString(tagType : string, attr : string, attrValue : string) : string {
+        let tag = '';
+        if (tagType === 'script') {
+            tag = `<script ${attr}='${attrValue}'></script>`;
```

Comment:
Generating strings and glueing them together feels slightly too low level what about using
`document.createElement('script');`
?

Reply:
I am trying to avoid using term `document` to avoid any probably SSR issues. wdyt?

Reply:
@CezCz, you were off when we had that conversation.

The library must be isomorphic (usable in the browser as well as on Node.js) and Node.js doesn't embark, a virtual DOM module, API, window, or document objects. See [1]

Libraries such as jsDom are trying to fill that gap. See [0]

[0] https://www.npmjs.com/package/jsdom
[1] https://stackoverflow.com/a/6657243

@CezCz what is your perspective?

Reply:
After some conversation, we challenged the use of string over DOMElements.

@rachitkumar13 could you please compare the architectural feasibility and elegance of the following solutions:

* Return the DOMElements as a string, and expect the project team to use a library (or custom code) to convert that string to HTMLElements and then add these to the DOM
* Depend on jsDom (or similar modules) in the backend, always return DOMElements, and expect the HTMLElements added to the DOM by the project team

Note: I already asked and I/O Runtime would support the use of jsDom